floater/generators.py

```diff
@@ -361,18 +361,27 @@ def npart_to_2D_array(self, ds1d):
         frame_dict.update(var_dict)
         frame = pd.DataFrame(frame_dict)
         framei = frame.set_index('index')
+        framei.columns = var_list
         if self.model_grid is not None:
             ocean_bools = self.ocean_bools
         else:
             ocean_bools = np.zeros(Nt, dtype=bool)==False
         framei.loc[ocean_bools==True] = df.values.astype(np.float32)
         framei.loc[ocean_bools==False] = np.float32('nan')
         data_vars = {}
+        dim_list = list(ds1d.dims)
+        dim_list.remove('npart')
+        dim_len = len(dim_list)
+        new_shape = (1,)*dim_len + (Ny, Nx)
+        new_dims = dim_list + ['lat', 'lon']
```

Comment:
This this logic, `lat` and `lon` will always be the last dimensions. That is okay with me!